fluster/test_suite.py

```diff
@@ -539,8 +548,11 @@ def run(self, ctx: Context) -> Optional["TestSuite"]:
 
     def generate_tests(self, ctx: Context) -> List[Test]:
         """Generate the tests for a decoder"""
-        tests = []
+        tests: List[Test] = []
         test_vectors_run = {}
+        if ctx.decoder.codec == Codec.MPEG2_VIDEO:
+            self.test_method = TestMethod.PIXEL
+            ctx.keep_files = True
```

Comment:
You shouldn't modify `keep_files` here since you need to honor the user's configuration and delete the generated output files after the test finishes. PixelComparisonTest will call `Decoder.decode` with `keep_files=True` and after the comparison checks finished, it will delete the output files based on `keep_files`